src/posix/platform/misc.c

```diff
@@ -28,46 +28,32 @@
 
 #include "platform-posix.h"
 
+#include <setjmp.h>
 #include <unistd.h>
 
 #include <openthread/platform/misc.h>
 
 #include "openthread-system.h"
 
-extern int    gArgumentsCount;
-extern char **gArguments;
+extern jmp_buf gResetJump;
 
 static otPlatResetReason   sPlatResetReason = OT_PLAT_RESET_REASON_POWER_ON;
 bool                       gPlatformPseudoResetWasRequested;
 static otPlatMcuPowerState gPlatMcuPowerState = OT_PLAT_MCU_POWER_STATE_ON;
 
 void otPlatReset(otInstance *aInstance)
 {
-    int i = 0;
-    // Restart the process using execvp.
-    char *argv[gArgumentsCount + 1];
-
 #if OPENTHREAD_PLATFORM_USE_PSEUDO_RESET
     gPlatformPseudoResetWasRequested = true;
     sPlatResetReason                 = OT_PLAT_RESET_REASON_SOFTWARE;
 
 #else // elif OPENTHREAD_PLATFORM_USE_PSEUDO_RESET
 
-    for (i = 0; i < gArgumentsCount; ++i)
-    {
-        argv[i] = gArguments[i];
-    }
-
-    argv[gArgumentsCount] = NULL;
-
     otSysDeinit();
     platformUartRestore();
 
-    alarm(0);
-
-    execvp(argv[0], argv);
-    perror("reset failed");
-    exit(EXIT_FAILURE);
+    longjmp(gResetJump, 1);
```

Comment:
Will this ensure all the RAM variables in OT core are cleared and re-initialized?
Reset command can be used as a way to clear out anything in RAM.

Reply:
This jump back to main.c, where exec are called. From the man pages, the whole process are restarted. See http://www.man7.org/linux/man-pages/man2/execve.2.html